lib/daru/accessors/nmatrix_wrapper.rb

```diff
@@ -32,47 +32,47 @@ def [] *index
         return @data[*index] if index[0] < @size
         nil
       end
- 
+
       def []= index, value
-        raise ArgumentError, "Index #{index} does not exist" if 
-          index > @size and index < @data.size
+        raise ArgumentError, "Index #{index} does not exist" if
+          index > @size && index < @data.size
```

Comment:
(Already answered this somewhere) 
Just because its like Ruby works. `&&` is logical and, `and` is control-flow and; it differs in all possible ways, including operator precedence.